File: app/src/main/java/aws/apps/usbDeviceEnumerator/ui/usbinfo/fragments/FragmentFactory.java
package aws.apps.usbDeviceEnumerator.ui.usbinfo.fragments;

import androidx.fragment.app.Fragment;
import uk.co.alt236.usbdeviceenumerator.sysbususb.SysBusUsbDevice;

public final class FragmentFactory {

    public static Fragment getFragment(String usbKey) {
        return InfoFragmentAndroid.create(usbKey);
    }

    public static Fragment getFragment(SysBusUsbDevice usbDevice) {
        return InfoFragmentLinux.create(usbDevice);
    }
}

File: app/src/main/java/aws/apps/usbDeviceEnumerator/ui/usbinfo/fragments/FragmentFactory.kt
package aws.apps.usbDeviceEnumerator.ui.usbinfo.fragments

import androidx.fragment.app.Fragment
import aws.apps.usbDeviceEnumerator.ui.main.list.UiUsbDevice
import aws.apps.usbDeviceEnumerator.ui.usbinfo.fragments.android.InfoFragmentAndroid
import aws.apps.usbDeviceEnumerator.ui.usbinfo.fragments.linux.InfoFragmentLinux
import uk.co.alt236.usbdeviceenumerator.sysbususb.SysBusUsbDevice
import javax.inject.Inject

class FragmentFactory @Inject constructor() {

    fun getFragment(device: UiUsbDevice): Fragment {
        return when (device) {
            is UiUsbDevice.AndroidUsb -> getFragment(device.key)
            is UiUsbDevice.SysUsb -> getFragment(device.device)
        }
    }

    fun getFragment(androidKey: String): Fragment = InfoFragmentAndroid.create(androidKey)
    fun getFragment(device: SysBusUsbDevice) = InfoFragmentLinux.create(device)

}
